src/SAWScript/Crucible/LLVM/ResolveSetupValue.hs

```diff
@@ -522,3 +539,116 @@ equalValsPred cc v1 v2 = go (v1, v2)
+                remaining_bits_size_term <- liftIO $ scNat saw_context $
+                  (byte_count - byte_index - 1) * 8
+                saw_byte_term <- liftIO $ scSlice
+                  saw_context
+                  bit_type_term
+                  byte_index_term
+                  byte_size_term
+                  remaining_bits_size_term
+                  saw_term
+
+                let byte_storage_index = case endianess of
+                      Crucible.BigEndian -> byte_index
+                      Crucible.LittleEndian -> byte_count - byte_index - 1
+                let vector_index = (fromIntegral offset) + (fromIntegral byte_storage_index)
+                modify' $ \vector -> vector V.// [(vector_index, saw_byte_term)]
+            else
+              modify' $ \vector -> vector V.// [(fromIntegral offset, saw_term)]
+          | otherwise -> fail $ "unexpected bit count: " ++ show size
+
+        Cryptol.TVSeq size element_cryptol_type -> do
+          element_storage_size <- liftIO $
+            Crucible.storageTypeSize <$> toLLVMStorageType
+              data_layout
+              element_cryptol_type
+
+          forM_ [0 .. (size - 1)] $ \element_index -> do
+            size_term <- liftIO $ scNat saw_context $ fromInteger size
+            elem_type_term <- liftIO $ importType
+              saw_context
+              emptyEnv
+              (Cryptol.tValTy element_cryptol_type)
+            index_term <- liftIO $ scNat saw_context $ fromInteger element_index
+            inner_saw_term <- liftIO $ scAt
+              saw_context
+              size_term
+              elem_type_term
+              saw_term index_term
+            setBytes
+              element_cryptol_type
+              inner_saw_term
+              (offset + (fromInteger element_index) * element_storage_size)
+
+        Cryptol.TVTuple tuple_element_cryptol_types -> do
+          (Crucible.Struct field_storage_types) <- liftIO $
+            Crucible.storageTypeF <$> toLLVMStorageType data_layout cryptol_type
+
+          V.forM_ (V.izipWith (,,) field_storage_types (V.fromList tuple_element_cryptol_types)) $
+            \(field_index, field_storage_type, tuple_element_cryptol_type) -> do
+              inner_saw_term <- liftIO $ scTupleSelector
+                saw_context
+                saw_term
+                (field_index + 1)
+                (length tuple_element_cryptol_types)
+              setBytes
+                tuple_element_cryptol_type
+                inner_saw_term
+                (offset + (Crucible.fieldOffset field_storage_type))
+
+        _ -> fail $ "unexpected cryptol type: " ++ show cryptol_type
+
+  case ttSchema typed_term of
+    Cryptol.Forall [] [] cryptol_type -> do
+      let evaluated_type = (Cryptol.evalValType Map.empty cryptol_type)
+      storage_size <- Crucible.storageTypeSize <$> toLLVMStorageType
+        data_layout
+        evaluated_type
+      padding_byte_term <- scBvConst saw_context 8 0xff
```

Comment:
Why this constant here? Does it just not matter what the padding bytes are set to?

Reply:
I just put something, the padding bytes are never accessed